### PR TITLE
feat!: Set merge mode while creating table in influx handler

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This PR sets the `merge_mode` to `last_non_null` for tables created automatically by the influx protocol handler. Now tables created by the influx handler have the same merge mode as `measurements` in InfluxDB.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Improved handling of row inserts with various table types including `LastNonNull`.
  - Added support for dynamically creating tables based on specified characteristics.
  - Enhanced functionality for inserting and querying InfluxDB-like data.

- **Bug Fixes**
  - Corrected retrieval and insertion processes to ensure accurate data handling for multiple hosts in InfluxDB scenarios.

- **Refactor**
  - Streamlined methods related to row inserts for better clarity and maintainability.
  - Updated enum and function names to reflect their roles more accurately.

- **Tests**
  - Enhanced integration tests to verify correct data retrieval for multi-host scenarios in InfluxDB.

### CODE REVIEW

**Comment by evenyag** [comment]
@coderabbitai summary

File: src/operator/src/insert.rs [hunk 1]
```diff
@@ -612,69 +633,59 @@ impl Inserter {
...
+                .insert(k.to_string(), v.to_string());
+        }
         let res = statement_executor
             .create_table_inner(create_table_expr, None, ctx.clone())
             .await;
 
         match res {
             Ok(table) => {
-                info!("Successfully created a log table {}", table_ref);
+                info!("Successfully created table {}", table_ref);
```
Comment (killme2008) [inline src/operator/src/insert.rs:684]: Better to print the chosen table options into the log too.

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/operator/src/insert.rs [hunk 2]
```diff
@@ -66,10 +67,17 @@ pub struct Inserter {
...
+/// Hint for the table type to create automatically.
+#[derive(Clone)]
 enum AutoCreateTableType {
+    /// A logical table with the physical table name.
     Logical(String),
+    /// A physical table.
     Physical,
+    /// A log table which is append-only.
     Log,
+    /// A table that merge rows by `last_non_null` strategy.
```
Comment (killme2008) [inline src/operator/src/insert.rs:79]:
```suggestion
    /// A table that merges rows by `last_non_null` strategy.
    /// Currently available only for InfluxDB Line Protocol.
```

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**